modules/api/src/main/java/org/apache/fluo/api/data/Bytes.java

```diff
@@ -356,29 +357,44 @@ public static final Bytes of(String s, Charset c) {
   /**
    * Checks if this has the passed prefix
    * 
-   * @param prefix
+   * @param prefix is a Bytes object to compare to this
    * @return true or false
+   * @since 1.1.0
    */
   public boolean startsWith(Bytes prefix) {
-    for (int i = 0; i < prefix.length; i++) {
-      if (this.byteAt(i) != prefix.byteAt(i))
-        return false;
+    Objects.requireNonNull(prefix, "startWith(Bytes prefix) cannot have null parameter");
+
+    if (prefix.length > this.length) {
+      return false;
+    } else {
+      for (int i = 0; i < prefix.length; i++) {
```

Comment:
To avoid repeatedly adding offsets, could do something like the following.  

```java
int end = this.offset + prefix.length;
for (int i  = this.offset, j = prefix.offset; i < end; i++,j++) 
``` 

I am not completely sure about the syntax.